src/utils/RPGUtil.ts

```diff
@@ -1,24 +1,25 @@
 import { ferreiro } from '../structures/Rpgs/items.json';
 import { IInventoryUser, IUserRpgSchema } from './Types';
 
+type ResultItem = IInventoryUser & { amount: number };
 export default class RPGUtil {
-  static countItems(items: IInventoryUser[]): Array<IInventoryUser & { amount: number }> {
-    return items.reduce((p, v) => {
-      const exists = p.findIndex((x) => x.name === v.name);
-      if (exists !== -1) {
-        p[exists].amount++;
-        return p;
-      }
-      return [
-        ...p,
-        {
-          name: v.name,
+  static countItems(items: IInventoryUser[]): Array<ResultItem> {
+    const result = items.reduce((acc, cur) => {
+      const item = acc.get(cur.name);
+
+      if (item) {
+        item.amount++;
```

Comment:
Almeidx commented about the item.amount +=1;

doesn't eslint see it as a problem?

Reply:
nop

Reply:
Can we merge it?